src/common/alignment.h

```diff
@@ -37,4 +40,80 @@ constexpr bool IsWordAligned(T value) {
     return (value & 0b11) == 0;
 }
 
+template <typename T, std::size_t Align = 16>
+class AlignmentAllocator {
+public:
+    typedef T value_type;
+    typedef std::size_t size_type;
+    typedef std::ptrdiff_t difference_type;
+
+    typedef T* pointer;
+    typedef const T* const_pointer;
+
+    typedef T& reference;
+    typedef const T& const_reference;
+
+public:
+    inline AlignmentAllocator() throw() {}
+
+    template <typename T2>
+    inline AlignmentAllocator(const AlignmentAllocator<T2, Align>&) throw() {}
+
+    inline ~AlignmentAllocator() throw() {}
+
+    inline pointer adress(reference r) {
```

Comment:
```suggestion
    inline pointer address(reference r) {
```

Ditto below.

Note that you previously told me you tested this interface on Discord. This is proving to be not the case, given it's a central part of the allocator interface (until C++20). Particularly given the implementation itself is incorrect, as described below.